tests/bluetooth/tester/src/mesh.c

```diff
@@ -0,0 +1,305 @@
+/* mesh.c - Bluetooth MESH Tester */
+
+/*
+ * Copyright (c) 2017 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <bluetooth/bluetooth.h>
+
+#include <errno.h>
+#include <bluetooth/mesh.h>
+#include <misc/byteorder.h>
+#include "bttester.h"
+
+#define CONTROLLER_INDEX 0
+
+#define CID 0xffff
+
+/* Configure provisioning data */
+static u8_t dev_uuid[16];
+static u8_t static_auth[16];
+
+/* Provision node data */
+static u8_t net_key[16];
+static u16_t net_key_idx;
+static u8_t flags;
+static u32_t iv_index;
+static u32_t seq_num;
+static u16_t addr;
+static u8_t dev_key[16];
+
+static void supported_commands(u8_t *data, u16_t len)
+{
+	u8_t cmds[1];
+	struct mesh_read_supported_commands_rp *rp = (void *) cmds;
+
+	memset(cmds, 0, sizeof(cmds));
+
+	tester_set_bit(cmds, MESH_READ_SUPPORTED_COMMANDS);
+	tester_set_bit(cmds, MESH_CONFIG_PROVISIONING);
+	tester_set_bit(cmds, MESH_PROVISION_NODE);
+	tester_set_bit(cmds, MESH_INIT);
+	tester_set_bit(cmds, MESH_RESET);
+	tester_set_bit(cmds, MESH_INPUT_NUMBER);
+	tester_set_bit(cmds, MESH_INPUT_STRING);
+
+	tester_send(BTP_SERVICE_ID_MESH, MESH_READ_SUPPORTED_COMMANDS,
+		    CONTROLLER_INDEX, (u8_t *) rp, sizeof(cmds));
+}
+
+static struct bt_mesh_cfg_srv cfg_srv = {
+	.relay = BT_MESH_RELAY_DISABLED,
+	.beacon = BT_MESH_BEACON_ENABLED,
+	.frnd = BT_MESH_FRIEND_NOT_SUPPORTED,
+	.gatt_proxy = BT_MESH_GATT_PROXY_NOT_SUPPORTED,
+	.default_ttl = 7,
+
+	/* 3 transmissions with 20ms interval */
+	.net_transmit = BT_MESH_TRANSMIT(2, 20),
+	.relay_retransmit = BT_MESH_TRANSMIT(2, 20),
+};
+
+static struct bt_mesh_health_srv health_srv = {
+};
+
+static struct bt_mesh_model_pub health_pub = {
+	.msg = BT_MESH_HEALTH_FAULT_MSG(0),
+};
+
+static struct bt_mesh_model root_models[] = {
+	BT_MESH_MODEL_CFG_SRV(&cfg_srv),
+	BT_MESH_MODEL_HEALTH_SRV(&health_srv, &health_pub),
+};
+
+static struct bt_mesh_elem elements[] = {
+	BT_MESH_ELEM(0, root_models, BT_MESH_MODEL_NONE),
+};
+
+static void link_open(bt_mesh_prov_bearer_t bearer)
+{
+	SYS_LOG_DBG("Provisioning link opened");
+}
+
+static void link_close(bt_mesh_prov_bearer_t bearer)
+{
+	SYS_LOG_DBG("Provisioning link closed");
+}
+
+static int output_number(bt_mesh_output_action_t action, uint32_t number)
+{
+	struct mesh_out_number_action_ev ev;
+
+	ev.action = sys_cpu_to_le16(action);
+	ev.number = sys_cpu_to_le32(number);
+
+	tester_send(BTP_SERVICE_ID_MESH, MESH_EV_OUT_NUMBER_ACTION,
+		    CONTROLLER_INDEX, (u8_t *) &ev, sizeof(ev));
+
+	return 0;
+}
+
+static int output_string(const char *str)
+{
+	u8_t string_len = strlen(str);
+	u8_t buf[sizeof(struct mesh_out_string_action_ev) + string_len];
```

Comment:
lets not use VLAs, since this shoudn't be larger than output_size we can truncate or return error in this happen. And just arbitrarily limit size (eg 16 bytes since thats max auth size anyway)

Reply:
with tx net_buf it'll be not needed (https://github.com/zephyrproject-rtos/zephyr/pull/4314/commits/fc491bdf78ed67d7a706646a053f319cff794bdf) so lets wait for the merge.